graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetService.java

```diff
@@ -38,6 +38,13 @@
     Optional<IndexSetConfig> get(String id);
 
     /**
+     * Retrieve the default index set.
+     *
+     * @return A filled {@link Optional} with the default index set, an empty {@link Optional} if there is no default.
```

Comment:
The default index set should be there. If it's not then

- Content pack imports that contain streams will fail
- The old indices REST API endpoints without an index set parameter will fail
- The `V20161116172200_CreateDefaultStreamMigration` will fail

If we change that to not returning an optional and throwing an error when the default index set is absent, we also need a migration to migrate the default flag from the `index_sets` collection to the cluster config object.